kubemarine/patches/p1_kubeadm_flags.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021-2023 NetCracker Technology Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from textwrap import dedent
+from io import StringIO
+
+from kubemarine.core.action import Action
+from kubemarine.core.patch import RegularPatch
+from kubemarine.core.resources import DynamicResources
+from kubemarine import kubernetes
+
+class TheAction(Action):
+    def __init__(self):
+        super().__init__("Updpate kubeadm_flags after cri upgrade")
+
+    def run(self, res: DynamicResources):
+        cluster = res.cluster()
+        path = 'plugins."io.containerd.grpc.v1.cri"'
+        target_kubernetes_version = cluster.inventory["services"]["kubeadm"]["kubernetesVersion"]
+        kubernetes_nodes = cluster.make_group_from_roles(['control-plane', 'worker'])
+        pause_version = cluster.globals['compatibility_map']['software']['pause'][target_kubernetes_version]['version']
+        sandbox = cluster.inventory["services"]["cri"]['containerdConfig'][path]["sandbox_image"]
+        param_begin_pos = sandbox.rfind(":")
+        sandbox = sandbox[:param_begin_pos] + ":" + str(pause_version)
+        if "cri" in cluster.raw_inventory["services"] and "sandbox_image" in cluster.raw_inventory["services"]["cri"]["containerdConfig"][path]:
+            print("Skipping the patch for Kubeadm_Flags")
+            exit(0)
```

Comment:
1. Custom inventory may contain `services.cri` section, but not `services.cri.containerdConfig.plugins."io.containerd.grpc.v1.cri"`. In this case it will fail.
2. Please do not `exit`. See https://github.com/Netcracker/KubeMarine/blob/0.19.0/kubemarine/core/action.py#L40
3. Please use logging approach instead of `print`.

Reply:
1. Added the condition to handle if inventory do not contain `services.cri.containerdConfig.plugins."io.containerd.grpc.v1.cri"`.
2. Also updated error msg to use `cluster.log.debug` instead of `print`